proposals/lip-define_new_transaction_schema.md

```diff
@@ -0,0 +1,241 @@
+| `Transaction` | object | Must follow the `transactionSchema` schema with the only difference that `params` property is not serialized and contains the values of parameters of `paramsSchema` for the corresponding transaction. | An object representing a non-serialized transaction.                                        |
+
+### JSON Schema
+
+Transactions are serialized using `transactionSchema` given below.
+
+
+```java
+transactionSchema = {
+    "type": "object",
+    "required": [
+        "module",
+        "command",
+        "nonce",
+        "fee",
+        "senderPublicKey",
+        "params",
+        "signatures"
+    ],
+    "properties": {
+        "module": {
+            "dataType": "string",
+            "fieldNumber": 1
+        },
+        "command": {
+            "dataType": "string",
+            "fieldNumber": 2
+        },
+        "nonce": {
+            "dataType": "uint64",
+            "fieldNumber": 3
+        },
+        "fee": {
+            "dataType": "uint64",
+            "fieldNumber": 4
+        },
+        "senderPublicKey": {
+            "dataType": "bytes",
+            "length": ED25519_PUBLIC_KEY_LENGTH,
+            "fieldNumber": 5
+        },
+        "params": {
+            "dataType": "bytes",
+            "fieldNumber": 6
+        },
+        "signatures": {
+            "dataType": "array",
+            "items": {
+                "dataType": "bytes",
+                "length": ED25519_SIGNATURE_LENGTH 
+            },
+            "fieldNumber": 7
+        }
+    }
+}
+```
+
+
+#### Validation
+
+For a transaction `trs` to be valid, it must satisfy the following:
+
+* `trs` must follow the `transactionSchema`. 
+* `trs.params` is of length less than or equal to `MAX_PARAMS_SIZE` .
+
+The validity of each transaction is verified in the [static validation stage][lip-0055#block-processing] of the block processing. 
+
+
+### Serialization
+
+The serialization of an object of type `Transaction` is described in the following pseudocode.
+
+```python
+def encode(transactionSchema: LiskJSONSchema, trs: Transaction) -> bytes:
+    trs.params = encode(paramsSchema,trs.params)
+    return encode(transactionSchema,trs)
+```
+
+### Deserialization
+
+Consider a binary message `trsMsg`, corresponding to a serialized transaction. The deserialization procedure is as follows:
+
+```python
+def decode(transactionSchema: LiskJSONSchema, trsMsg: bytes) -> Transaction: 
+    trsData = decode(transactionSchema,trsMsg)
+    trsData.params = decode(paramsSchema,trsData.params)
+    return trsData
+```
+
+### Transaction signature calculation
+Consider a data structure `unsignedTrsData` representing a valid `Transaction` object in which the signatures array is initialized to the default value (an empty array). The following function calculates a signature of the object on a certain chain with secret key `sk`. 
+
+```python
+def signTransaction(sk: PrivateKeyEd25519, unsignedTrsData: Transaction) -> SignatureEd25519:
```

Comment:
Maybe we could call the function `computeTransactionSignature`. The current function name suggests that a signed transaction is returned, similar to the function (signTransaction)[https://github.com/LiskHQ/lisk-sdk/blob/ef942e74ea55f44db1af3a720093969d6a42194b/elements/lisk-transactions/src/sign.ts#L77] in the Lisk SDK.

Reply:
Name changed. `networkIdentifier` added as input parameter. 
I kept the `Transaction` object as input argument and recover the paramsSchema from the (module,command) tuple.

Reply:
Looks good.